### PR TITLE
Add docs for AWS per-source roles/profiles

### PR DESCRIPTION
**Pull request by quodlibetor**
This documents #9894

### CODE REVIEW

File: doc/user/layouts/partials/specifying-aws-credentials.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+`secret_access_key` | A valid [secret access key][access] for the AWS resource.
+`token` (optional) | The session token associated with the credentials, if the credentials are temporary
+
+Alternatively, you can specify an profile to assume. This named profile must exist within either the `credentials` or `config` file:
+
+Field | Description
+------|------------
+`profile` | An AWS config profile to assume.
+
+The following `WITH` options can be set with either static credentials, a profile, or while depending on the environment for credentials.
```
Comment (ruf-io) [inline doc/user/layouts/partials/specifying-aws-credentials.html:34]: Did you mean `either static credentials, a profile, or alone`?
```suggestion
The following `WITH` options can be set with either static credentials, a profile, or alone depending on the environment for credentials.
```

File: doc/user/layouts/partials/specifying-aws-credentials.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+1. Statically provided as part of the `WITH` block in source declaration. These credentials will be
+   written to disk in plain text as well as being easily exposed via introspection commands and so
+   this technique is only recommended for experimentation and development.
+2. Environment variables: `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`,
+   [`AWS_WEB_IDENTITY_TOKEN_FILE`][web-identity], and [many more][env-vars].
+3. `credential_process` command in the AWS config file, by default located at `~/.aws/config`.
+4. AWS credentials file and profile files. By default located at `~/.aws/config` and
+   `~/.aws/credentials` , respectively.
+5. The [IAM instance profile][instance-profile] provided by the Instance Metadata Service. Will
+   only work if running on an EC2 instance with an instance profile.
```
Comment (ruf-io) [inline doc/user/layouts/partials/specifying-aws-credentials.html:14]:
```suggestion
5. The [IAM instance profile][instance-profile] provided by the Instance Metadata Service. This
   will only work if running on an EC2 instance with an instance profile.
```

**Review by morsapaes — COMMENTED** [review]
Thanks, @quodlibetor! Left a few minor comments. Is there a logical way to break down the RR diagrams so they don't overflow?

Screenshot 2022-02-09 at 15 20 41

File: doc/user/layouts/partials/create-source/connector/kinesis/with-options.html [hunk 3]
```diff
@@ -1,4 +1,6 @@
-{{ partial (printf "aws-credentials-with-options") . -}}
+`timestamp_frequency_ms`| `int` | Default: `1000`. Sets the timestamping frequency in `ms`. Reflects how frequently the source advances its timestamp. This measure reflects how stale data in views will be. Lower values result in more-up-to-date views but may reduce throughput.
+
+{{ partial (printf "specifying-aws-credentials") . -}}
```
[the remark this answers is not in the document]
Comment (quodlibetor) [reply doc/user/layouts/partials/create-source/connector/kinesis/with-options.html]: Okay I've tried this, let me know what you think of the new placement.

**Comment by quodlibetor** [comment]
@morsapaes 
> Is there a logical way to break down the RR diagrams so they don't overflow?

I could do something like this?

image

Breaking the static creds out like that makes it a little bit harder to follow I think, but it's technically still correct. Possibly for some people it's easier to follow since it's less noisy? It seems fine to me, though.

**Comment by morsapaes** [comment]
Thanks for the updates, @quodlibetor! Good to merge from my side.

**Review by morsapaes — APPROVED** [review]

**Comment by quodlibetor** [comment]
@ruf-io lmk how you feel about these changes.

**Comment by benesch** [comment]
I'm going to assume that no news is good news from @ruf-io. Merging so that I can link to this in the v0.20.0 release notes.

**Pull request merged**